### PR TITLE
perf: Cache ProjectOwnership reads for five minutes

### PR DESCRIPTION
**Pull request by markstory**
Apply our default record cache to project ownership read queries. This should remove more load from postgres. This shouldn't introduce any delays in project ownership rules taking effect as updates will also update the cache data.

cc @JTCunning

### CODE REVIEW

File: src/sentry/models/projectownership.py [hunk 1]
```diff
@@ -16,6 +16,8 @@
 class ProjectOwnership(Model):
     __core__ = True
 
+    objects = BaseManager(cache_fields=["project_id", "pk"])
```
Comment (JTCunning) [inline src/sentry/models/projectownership.py]: Should this be `id` instead of `pk`?

Comment (mattrobenolt) [reply src/sentry/models/projectownership.py]: pk is an alias for id.

**Comment by mattrobenolt** [comment]
I feel like this is a very strong contender for a negative cache. Since more often than not, there would be no ownership rule.

**Comment by markstory** [comment]
Would we want both a negative cache and cached reads on hits?

**Comment by mattrobenolt** [comment]
I’d say so. I just don’t think a positive cache will do a ton since it probably doesn’t have many hits.

**Comment by JTCunning** [comment]
@markstory: I'd say definitely. The same thing occurs with GroupSnooze which is why https://github.com/getsentry/sentry/pull/15055 circumvents the manager cache.

**Comment by markstory** [comment]
Updated with the negative cache.

**Review by wedamija — APPROVED** [review]

**Pull request merged**